File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples/VB/WPFListBox-VB/WPFComponents/AdvWorks.vb



Partial Class AdvWorksDataContext
   Public Function GetAllCustomers() As IOrderedQueryable(Of Customer)
      Dim items = From cust In Me.Customers _
                    Order By cust.LastName _
                   Select cust

      Return items
   End Function

   Public Function GetAllCustomersAsList() As IEnumerable(Of Customer)
      Dim items = From cust In Me.Customers _
                   Order By cust.LastName _
                   Select cust

      Return items.ToList()
   End Function
End Class

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/CSharp/WPFListBox/WPFComponents/AdvWorks.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace WPFComponents
{
  partial class AdvWorksDataContext
  {
    public IOrderedQueryable<Customer> GetAllCustomers()
    {
      var items = from cust in this.Customers
                  orderby cust.LastName
                  select cust;

      return items;
    }

    public IEnumerable<Customer> GetAllCustomersAsList()
    {
      var items = from cust in this.Customers
                  orderby cust.LastName
                  select cust;

      return items.ToList();
    }
  }
}
